app/controllers/qa_controller.rb

```diff
@@ -0,0 +1,136 @@
+require 'gds_api/helpers'
+
+class QaController < ApplicationController
+  layout "finder_layout"
+  include GdsApi::Helpers
+
+  def show
+    return error_not_found unless ENV["FINDER_FRONTEND_ENABLE_QA"]
+
+    @content_item = raw_finder
+    fetch_breadcrumbs
+  end
+
+private
+
+  def qa_config
+    @qa_config ||= YAML.load_file("lib/#{request.path.tr('-', '_')}.yaml")
+  end
+
+  def raw_finder
+    @raw_finder ||= Services.content_store.content_item(qa_config["finder_base_path"])
+  end
+
+  def facets
+    @facets ||= raw_finder["details"]["facets"].select do |facet|
+      facet["type"] == "text" && facet["filterable"] && qa_config["pages"][facet["key"]]["show_in_qa"]
+    end
+  end
+
+  def page
+    params.permit(:page)
+    params[:page] = 1 if params[:page].nil?
+    params[:page].to_i.clamp(1, facets.length)
+  end
+
+  def fetch_breadcrumbs
+    parent_content_item = Services.content_store.content_item(qa_config["finder_base_path"])
+    @breadcrumbs = FinderBreadcrumbsPresenter.new(parent_content_item, @content_item)
+  end
```

Comment:
Why not
```
def fetch_breadcrumbs
  @breadcrumbs = FinderBreadcrumbsPresenter.new(raw_finder, @content_item)
end
```
?

Reply:
Also, isn't `@content_item` the same as `raw_finder`? Aren't you passing the same argument twice?

Reply:
If two arguments are expected, the method could be
```
def fetch_breadcrumbs
  @breadcrumbs = FinderBreadcrumbsPresenter.new(raw_finder, raw_finder)
end
```
This way, `@content_item = raw_finder` could also be removed from the `show` action (assuming it was only needed here).

Reply:
Fixed